test/stub-test.js

```diff
@@ -285,6 +285,65 @@ describe("stub", function () {
         });
     });
 
+    describe(".resolvesThis", function () {
+        afterEach(function () {
+            if (Promise.resolve.restore) {
+                Promise.resolve.restore();
+            }
+        });
+
+        it("returns a promise resolves with this", function () {
+            var instance = {};
+            instance.stub = createStub.create();
+            instance.stub.resolvesThis();
+
+            return instance.stub().then(function (actual) {
+                assert.same(actual, instance);
+            });
+        });
+
+        var strictMode = (function () {
+            return this;
+        }()) === undefined;
+        if (strictMode) {
+            it("returns a promise resolves with undefined when detached", function () {
+                var stub = createStub.create();
+                stub.resolvesThis();
+
+                // Due to strict mode, would be `global` otherwise
+                return stub().then(function (actual) {
+                    assert.same(actual, undefined);
+                });
+            });
+        }
+
+        it("stub respects call", function () {
```

Comment:
Can this expectation be more specific, to make it easier to understand what "respects call" means?